src/lib.rs

```diff
@@ -45,15 +45,74 @@ pub struct Header {
     pub generator: u32,
 }
 
+/// Forces early fragment tests. In a standard situation if a driver determines that it is possible to
+/// switch on early depth test it will. Typical situations when early depth test is switched off:
+///   - Calling ```discard``` in a shader.
+///   - Writing to the depth buffer, unless ConservativeDepthTest is enabled.
+///
+/// Enabling this option will do 2 things:
+///   - Disables any discard and depth write.
+///   - Ensures that image load/store operations will only happen on fragments that pass the depth test.
+///
+/// SPIR-V: ExecutionMode EarlyFragmentTests
+/// In GLSL: layout(early_fragment_tests) in;
+/// HLSL: Attribute earlydepthstencil
+///
+/// For more, see:
+///   - https://www.khronos.org/opengl/wiki/Early_Fragment_Test#Explicit_specification
+///   - https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/sm5-attributes-earlydepthstencil
+#[derive(Clone, Copy, Debug, PartialEq, Default)]
+#[cfg_attr(feature = "serialize", derive(Serialize))]
+#[cfg_attr(feature = "deserialize", derive(Deserialize))]
+pub struct ForcedEarlyDepthTest;
```

Comment:
I wonder what the intention is for making this `Option<UnitStruct>` instead of just a `bool`?

Reply:
Changed to bool.